iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt5/Mqtt5IotHubConnection.java

```diff
@@ -206,7 +227,45 @@ else if (proxySettings.getProxy().type() == Proxy.Type.HTTP)
             IMqttToken token = this.mqttAsyncClient.connect(connectOptions);
             token.waitForCompletion(); // TODO timeout value?
 
-            //TODO send CONNECT message to MQTT broker so E4K knows this client has connected
+            try
+            {
+                log.debug("Sending the \"Opened a connection\" message to the MQTT broker");
+                E4KConnectionMessagePayload connectPayload = E4KConnectionMessagePayload.builder()
+                    .connectionState(E4KConnectionState.Connected)
+                    .modelId(this.config.getModelId())
+                    .mqttVersion(MQTT_VERSION_STRING)
+                    .deviceClientType(this.config.getProductInfo().getUserAgentString())
+                    .build();
+                byte[] serializedPayload = GSON.toJson(connectPayload).getBytes(StandardCharsets.UTF_8);
+                org.eclipse.paho.client.mqttv3.IMqttDeliveryToken token = this.mqttAsyncClient.publish(
```

Comment:
Does Mqtt5 have an object type similar to `mqttv3.IMqttDeliveryToken` here?

Reply:
Good catch. This was a copy paste error. I have pushed the fix and now this is an Mqtt5 object